### DISCUSSION TITLE
Is it possible to translate the search block?

### DISCUSSION THREAD

**Discussion by kutepoff45** [piccolo-orm/piccolo_theme#81] [Q&A] [answered]
Hello there. Thanks to all contributors for this theme.

I wanna translate the search block into russian, but I don't know how to do that. I've tried sphinx-intl tool, but I can't find msgid for words needed to be translated. I've also read your docs for this theme and haven't found any possible option to set up localization. Could somebody help me, pls?

![Снимок экрана 2024-07-16 205657](https://github.com/user-attachments/assets/d839759e-56ab-4732-b2d8-dc33660136cd)

**Comment by dantownsend** [comment, marked as answer]
There's a PR which wraps all translatable terms in the `_` function:

https://github.com/piccolo-orm/piccolo_theme/pull/60

I need to check out how other themes handle their translations.

The Sphinx Book theme stores them as JSON:

 * https://sphinx-book-theme.readthedocs.io/en/stable/contributing/translate.html
 * https://github.com/executablebooks/sphinx-book-theme/tree/master/src/sphinx_book_theme/assets/translations

And then runs a script to generate the `locales/**/*.po` files.

Here are the main docs for handling [translation](https://www.sphinx-doc.org/en/master/usage/advanced/intl.html), and it mentions [sphinx-intl](https://github.com/sphinx-doc/sphinx-intl).

I need to go through the docs and work out the workflow.

So, not a great answer, but I plan on fixing it once I understand how Sphinx handles translations.
